src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -649,23 +649,34 @@ public override void CloseAdditionalDocument(DocumentId documentId)
 
         public bool TryGetInfoBarData(DocumentId documentId, out IVsWindowFrame frame, out IVsInfoBarUIFactory factory)
         {
+            frame = null;
+            factory = null;
             if (documentId == null)
             {
-                frame = null;
-                factory = null;
-                return false;
+                var monitorSelectionService = ServiceProvider.GetService(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
+                object value = null;
+                if (monitorSelectionService != null &&
+                   (ErrorHandler.Succeeded(monitorSelectionService.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_DocumentFrame, out value)) ||
+                    ErrorHandler.Succeeded(monitorSelectionService.GetCurrentElementValue((uint)VSConstants.VSSELELEMID.SEID_WindowFrame, out value))))
```

Comment:
Is it right that we prefer the document if possible?  There isn't a way to just get whatever is focused?